lib/compact.js

```diff
@@ -544,6 +544,18 @@ api.compact = async ({
                 compactedItem[typeKey] = types;
                 break;
             }
+
+            // If compactedItem contains a single entry
+            // whose key maps to @id, recompact without @type
+            if(Object.keys(compactedItem).length === 1 && '@id' in expandedItem) {
+              compactedItem = await api.compact({
```

Comment:
Can we just get the alias for `@id` here and pull out the value `compactedItem[alias]` without having to run compact again?

Reply:
Not easily, as if reduced to a node reference, it can be turned into a string. If not, it remains a node object. We could conceivably implement that logic there, but that's what expand is for, and it should be pretty quick in that case.